### ISSUE TITLE
Siri cannot find Homebridge devices without failing first after reconnecting to WiFi

### ISSUE THREAD

**Issue by sooth** [nfarina/homebridge#404]
"Siri, Turn on the lights" "Sorry, I wasn't able to do that" "Siri, Turn on the lights" "Lights are on!". 

Every. Time. If I leave and come back, every, single, time, Siri fails to do anything the first time, then immediately works on the second try. Does anyone else have this problem? 

My setup:
iPhone 5 (JB), Ubuntu 14.04

GF's setup:
iPhone 6, OSX Yosemite.

**Comment by sooth** [comment]
Also, the irony that this is issue #404 and involves Siri not being able to find homebridge is hilarious.

**Comment by snowdd1** [comment]
I had this 404 issue (that name is sticky...) coming and going. What about third and fourth tries? I even had some "I can search the internet for it" in between, even though the recognized sentence was correct and spelled exactly the same to the times it worked.

I thought this was a localization issue, as I use Siri in German.

**Comment by sooth** [comment]
Siri in US English here, I would be doubtful about the localization causing this issue. It's more of a "I haven't bothered trying connect to any HomeKit device yet, and I can't quickly enough for the first try."

This very well may be some kind of issue with Siri/HomeKit itself in the searching/connecting. I wish I had the tools to determine this.. maybe connecting my phone to a bridged adapter running wireshark?

**Comment by Bart97** [comment]
Same here. I thought it was caused by the wifi signal being too weak outside of my house but I guess it's not.

**Comment by enriquez** [comment]
This seems to happen to me after not using Siri to control homebridge after some time. Once Siri "warms up" and fails the first time it works perfectly afterwards... even if I disconnect/reconnect to the wifi from my phone. It's after a few hours or so is when it takes Siri two tries.

I'm guessing the problem is on Siri's end.  While watching the homebridge log, I didn't see any activity during the failing case. I assume homebridge isn't getting the request at all.

**Comment by KhaosT** [comment]
@enriquez I guess it's related to Siri's privacy policy. Siri server will periodically wipe the server cache of user info (like accessories, homes, scenes) and when network condition isn't really good, Siri server may not be able to get an updated snapshot of the home info so it tells you it can't find any device right now...

**Comment by maddox** [comment]
Oh this must be what's happening to me as well.



> Siri server will periodically wipe the server cache of user info (like accessories, homes, scenes) and when network condition isn't really good, Siri server may not be able to get an updated snapshot of the home info so it tells you it can't find any device right now...

Wow, is this for real? Is this daily? It caches like THIS? on request? instead of in the background. That's sort of horrifying.

Is this how it works on certified HomeKit accessories or is this an issue with being "not" certified?

**Comment by KhaosT** [comment]
I'm not sure about frequency. When you make a request like "Turn on the lights in my bedroom", Siri will first send a request to HomeKit asking if there is a room named "bedroom", then after HomeKit responses, it will send a second request like "power -> on" for "lights" in "bedroom". After that, Siri (local/server) will cache the existence of "bedroom" for a period of time. Those requests take time and sometime the sync might not be completed before the request timeout.

I don't think certification has anything to do in this process. It's a sad consequence of protecting privacy I guess.

**Comment by maddox** [comment]
Any word on this. I can't believe it's actually a caching issue. It happens multiple times a day. 

I still can't believe that this is just how HomeKit works generally, even for those that use certified devices. There would be constant complaints not to mention the fact that people would generally regard HomeKit as broken and worthless, lol. But this def doesn't seem to be the case.

**Comment by KhaosT** [comment]
@maddox I haven't experience this problem for a while. Have you tried to turn off Siri (in Settings) and turn it back on?

**Comment by sooth** [comment]
I can't say I've had it happen much either, however, I haven't used it as much either.

My enthusiasm has also been tempered somewhat, so now instead of trying to say "I'm home!" right as I walk in, I'll walk in, set my gym bag down, take off my hat and coat, THEN say I'm home. I'm assuming this gives my phone much more time to detect it.

**Comment by chrischernoff** [comment]
I have the same issue pretty regularly. First try → Sorry. Second try → Ok, done!

**Comment by maddox** [comment]
@chrischernoff which plugins/accessories/platforms are you using in your Homebridge setup?

I'm personally almost exclusively using the [Home Assistant platform](https://github.com/nfarina/homebridge-legacy-plugins/blob/master/platforms/HomeAssistant.js).

**Comment by chrischernoff** [comment]
@maddox SmartThings and Nest

**Comment by tkoenig** [comment]
I was experiencing the same issues when running homebridge on a raspberry pi.
I remembered when running `nmap` on my macbook that the pi had two IP addresses assigned (I tried assigning a static IP, but did it wrong via `/etc/interfaces`). I have now made sure that the raspberry only uses [one IP address](https://www.raspberrypi.org/forums/viewtopic.php?p=798866#p798866).
I'm not sure if this really was the issue, but it seems to have fixed this for now (I just did this recently).

**Comment by sooth** [comment]
Interesting thought tkoenig, I'm sure my VM's have both an IPv4 and IPv6 address, I wonder if that is the source of some of the problem.. Apple seems to really have a love for IPv6 (AirPlay has lots of IPv6 only things IIRC) so I think I will try disabling IPv4 on this server and see how it goes.

**Comment by demetrio812** [comment]
Hello guys,
I've installed homebridge on my PI, everything works fine for the first 5-10 minutes then the HomeKit software (I used Elgato Eve) doesn't find any accessory anymore. I've followed the guide and homebridge is running automatically on start. The thing is that when I try to access to the http://<ip>:51826/accessories it works from my mac so it seems that the server is accessible.

I've tried Elgato Eva in an iPad and an iPhone, same results. Obviously they are in the same network.

I remove the persist folder, change the name, delete the home in the HomeKit settings, restart the app and it works again for another 5 minutes...

What can it be? My pi is connected via ethernet so no wifi power problems...

Thanks,
Dem

**Comment by rudders** [comment]
Does your Pi have wireless and ethernet enabled? i.e. two paths onto your network. I have had this kind of problem in the past and disabled wireless and have had no issues since.

**Comment by demetrio812** [comment]
Thanks for your answer, I don't have any wifi dongle, I'm trying anyway to see if this changes something...thanks

**Comment by damianromanowski** [comment]
This also happens to me from time-to-time. Most notably when I try to enable my 'good night' scene. She attempts to run it, says there are no devices, and on the second try works 99% of the time. Or weirder yet, she says the scene has finished running, but I still have lights on in parts of the house. Running it a second time works properly.

**Comment by demetrio812** [comment]
The problem for me is that everytime it stops working I will have to stop, remove persist, renova, remove the home from settings and start again (so I loose all my configuration).

I've just tried again, this time I've installed also another copy of homebridge on my Mac. HomeKit recognised both of them and I can see both on Discovery app. 

Again the Eve app in iOS doesn't see anymore any appliance. I'm not sure what it could be. I've also tried with an iPad, same results. They both have iOS 9.

What else can I try?

Thanks,
Dem
